src/dft/backends/rocfft/commit.cpp

```diff
@@ -274,31 +275,40 @@ class rocfft_commit final : public dft::detail::commit_impl<prec, dom> {
         std::unique_ptr<rocfft_plan_description_t, decltype(description_destroy)>
             description_destroyer(plan_desc, description_destroy);
 
-        std::array<std::size_t, 3> in_stride_indices{ 0, 1, 2 };
-        std::sort(&in_stride_indices[0], &in_stride_indices[dimensions],
-                  [&](std::size_t a, std::size_t b) { return in_strides[a] < in_strides[b]; });
-        std::array<std::size_t, 3> out_stride_indices{ 0, 1, 2 };
-        std::sort(&out_stride_indices[0], &out_stride_indices[dimensions],
-                  [&](std::size_t a, std::size_t b) { return out_strides[a] < out_strides[b]; });
+        std::array<std::size_t, 3> stride_a_indices{ 0, 1, 2 };
+        std::sort(&stride_a_indices[0], &stride_a_indices[dimensions],
+                  [&](std::size_t a, std::size_t b) {
+                      return stride_vecs.vec_a[a] < stride_vecs.vec_a[b];
+                  });
+        std::array<std::size_t, 3> stride_b_indices{ 0, 1, 2 };
+        std::sort(&stride_b_indices[0], &stride_b_indices[dimensions],
+                  [&](std::size_t a, std::size_t b) {
+                      return stride_vecs.vec_b[a] < stride_vecs.vec_b[b];
+                  });
         std::array<std::size_t, max_supported_dims> lengths_cplx = lengths;
         if (dom == dft::domain::REAL) {
             lengths_cplx[0] = lengths_cplx[0] / 2 + 1;
         }
         // When creating real-complex descriptions, the strides will always be wrong for one of the directions.
         // This is because the least significant dimension is symmetric.
         // If the strides are invalid (too small to fit) then just don't bother creating the plan.
-        const bool valid_forward =
+        const bool vec_a_valid_as_reals =
             dimensions == 1 ||
-            (lengths_cplx[in_stride_indices[0]] <= in_strides[in_stride_indices[1]] &&
+            (lengths_cplx[stride_a_indices[0]] <= stride_vecs.vec_a[stride_a_indices[1]] &&
              (dimensions == 2 ||
-              lengths_cplx[in_stride_indices[0]] * lengths_cplx[in_stride_indices[1]] <=
-                  in_strides[in_stride_indices[2]]));
-        const bool valid_backward =
+              lengths_cplx[stride_a_indices[0]] * lengths_cplx[stride_a_indices[1]] <=
+                  stride_vecs.vec_a[stride_a_indices[2]]));
+        const bool vec_b_valid_as_reals =
             dimensions == 1 ||
-            (lengths_cplx[out_stride_indices[0]] <= out_strides[out_stride_indices[1]] &&
+            (lengths_cplx[stride_b_indices[0]] <= stride_vecs.vec_b[stride_b_indices[1]] &&
              (dimensions == 2 ||
-              lengths_cplx[out_stride_indices[0]] * lengths_cplx[out_stride_indices[1]] <=
-                  out_strides[out_stride_indices[2]]));
+              lengths_cplx[stride_b_indices[0]] * lengths_cplx[stride_b_indices[1]] <=
+                  stride_vecs.vec_b[stride_b_indices[2]]));
```

Comment:
Is there no similar requirement as the above for the complex domain? Say self-explanatory flags like `vec_(a|b)_valid_as_hermitians` (to use rocfft's terms)?

Reply:
I find this code confusing, and it looks like I got it wrong the first time round with my understanding of it. I tried to modify the code such that it was the same as what already existed.

* `lengths_cplx` look like the storage required for the DFT in the complex domain.
* So we're checking if strides vs `lengths_complex`, we're checking that the strides are sufficient for writing the data in the complex domain.

To conclude, I've named this variable badly and misleadingly. It is valid for both real and complex domain DFTs. I will rename this in a new PR.

Reply:
I've updated the code to check both fwd and bwd domains in the new PR: https://github.com/oneapi-src/oneMKL/pull/549
Change: https://github.com/hjabird/oneMKL/blob/128ef68cd2ed35c6543ca163a8d79bd69b678748/src/dft/backends/rocfft/commit.cpp#L304